lib/model/content.dart

```diff
@@ -351,26 +351,51 @@ class ImageNodeList extends BlockContentNode {
 }
 
 class ImageNode extends BlockContentNode {
-  const ImageNode({super.debugHtmlNode, required this.srcUrl});
+  const ImageNode({
+    super.debugHtmlNode,
+    required this.srcUrl,
+    required this.thumbnailUrl,
+    required this.loading,
+  });
```

Comment:
Let's split this into two constructors:
```suggestion
  const ImageNode({
    super.debugHtmlNode,
    required String this.srcUrl,
    required String this.thumbnailUrl,
  }) : loading = false;
```

and then an `ImageNode.loading` that makes those other fields null.

That way the constructors enforce that there are these two mutually exclusive configurations.

Reply:
… Actually, though, scratch that — because I think we want to model this a bit differently.

Even if the thumbnail is still pending, that's no reason to stop the user from opening the lightbox. So let's always have the original URL, which is the one from the `a` element.